### PR TITLE
Part 1: Add `@medplum/health-gorilla-core`

### PR DESCRIPTION
**Pull request by mattlong**
Part 1 of 3 from #5205

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -54147,6 +54149,7 @@
     "packages/health-gorilla": {
       "name": "@medplum/health-gorilla",
       "version": "3.2.8",
+      "extraneous": true,
```
Comment (mattlong) [inline package-lock.json:54152]: Besides breaking glass and manually deleting this whole entry from `package-lock.json`, it doesn't seem like it's possible to get rid of the now gone `packages/health-gorilla` entry.

https://stackoverflow.com/questions/78435644/npm-remove-workspace-from-monorepo
https://github.com/npm/cli/issues/5463
https://github.com/npm/cli/pull/5478

File: packages/health-gorilla-core/esbuild.mjs [hunk 2]
```diff
@@ -7,13 +8,26 @@ import { writeFileSync } from 'fs';
...
   sourcemap: true,
+  external: [
+    '@mantine/core',
+    '@mantine/hooks',
+    '@mantine/notifications',
+    '@mantine/react',
+    '@medplum/core',
+    '@medplum/mock',
+    '@medplum/react',
+    'react',
```
Comment (codyebberson) [inline packages/health-gorilla-core/esbuild.mjs:29]: Most of these aren't necessary for `health-gorilla-core`, yeah?

File: packages/health-gorilla-core/package.json [hunk 3]
```diff
@@ -1,20 +1,9 @@
 {
-  "name": "@medplum/health-gorilla",
-  "version": "3.2.9",
+  "name": "@medplum/health-gorilla-core",
+  "version": "3.2.8",
+  "private": true,
   "description": "Medplum Health Gorilla SDK",
-  "homepage": "https://www.medplum.com/",
```
Comment (codyebberson) [inline packages/health-gorilla-core/package.json:5]: Any objections to keeping these values here?  I think someone told us once that these improve NPM ranking algorithm.

File: packages/health-gorilla-core/package.json [hunk 4]
```diff
@@ -1,20 +1,9 @@
...
-  },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/medplum/medplum.git",
-    "directory": "packages/health-gorilla"
-  },
-  "license": "Apache-2.0",
-  "author": "Medplum <hello@medplum.com>",
   "sideEffects": false,
-  "type": "module",
```
Comment (codyebberson) [inline packages/health-gorilla-core/package.json:17]: We're trying to move all projects to `module` over time.  Any reason not to here?

File: packages/health-gorilla-core/package.json [hunk 5]
```diff
@@ -27,7 +16,7 @@
   },
   "main": "dist/cjs/index.cjs",
   "module": "dist/esm/index.mjs",
-  "types": "dist/esm/index.d.ts",
+  "types": "dist/cjs/index.d.ts",
```
Comment (codyebberson) [inline packages/health-gorilla-core/package.json:19]: Same as `module` comment - although in this case they're probably the exact same content.

**Review by codyebberson — APPROVED** [review]

**Pull request merged**